crates/re_viewer/src/ui/recordings_panel.rs

```diff
@@ -34,58 +40,93 @@ fn recording_list_ui(ctx: &mut ViewerContext<'_>, ui: &mut egui::Ui) {
 
-    let desc = format_description!(version = 2, "[hour]:[minute]:[second]Z");
-    for store_db in &store_dbs {
-        let info = if let Some(store_info) = store_db.store_info() {
-            format!(
-                "{} - {}",
-                store_info.application_id,
-                store_info
-                    .started
-                    .to_datetime()
-                    .and_then(|dt| dt.format(&desc).ok())
-                    .unwrap_or("<unknown>".to_owned())
-            )
+    for (app_id, store_dbs) in store_dbs_map {
+        if store_dbs.len() == 1 {
+            let store_db = store_dbs[0];
+            if recording_ui(ctx.re_ui, ui, store_db, Some(app_id), active_recording).clicked() {
+                command_sender
+                    .send_system(SystemCommand::SetRecordingId(store_db.store_id().clone()));
+            }
         } else {
-            "<UNKNOWN>".to_owned()
-        };
-
-        if ctx
-            .re_ui
-            .list_item(info)
-            .with_icon_fn(|_re_ui, ui, rect, visuals| {
-                let color = if active_recording == Some(store_db.store_id()) {
-                    visuals.fg_stroke.color
-                } else {
-                    ui.visuals().widgets.noninteractive.fg_stroke.color
-                };
-
-                ui.painter()
-                    .circle(rect.center(), 4.0, color, egui::Stroke::NONE);
-            })
-            .show(ui)
-            .clicked()
-        {
-            command_sender.send_system(SystemCommand::SetRecordingId(store_db.store_id().clone()));
+            ctx.re_ui
+                .list_item(app_id)
+                .active(false)
+                .show_collapsing(ui, true, |_, ui| {
+                    for store_db in store_dbs {
+                        if recording_ui(ctx.re_ui, ui, store_db, None, active_recording).clicked() {
+                            command_sender.send_system(SystemCommand::SetRecordingId(
+                                store_db.store_id().clone(),
+                            ));
+                        }
+                    }
+                });
         }
     }
 }
 
+/// Show the UI for a single recording.
+///
+/// If an `app_id_label` is provided, it will be shown in front of the recording time.
+fn recording_ui(
+    re_ui: &re_ui::ReUi,
+    ui: &mut egui::Ui,
+    store_db: &re_data_store::StoreDb,
+    app_id_label: Option<&str>,
+    active_recording: Option<&re_log_types::StoreId>,
+) -> egui::Response {
+    let prefix = if let Some(app_id_label) = app_id_label {
+        format!("{app_id_label} - ")
+    } else {
+        String::new()
+    };
+
+    let name = if let Some(store_info) = store_db.store_info() {
+        store_info
+            .started
+            .to_datetime()
+            .and_then(|dt| dt.format(&TIME_FORMAT_DESCRIPTION).ok())
+            .unwrap_or("<unknown>".to_owned())
+    } else {
+        "<unknown>".to_owned()
```

Comment:
I think it is good if we let the user know _what_ is unknown

```suggestion
            .unwrap_or("<unknown time>".to_owned())
    } else {
        "<unknown time>".to_owned()
```